domain/src/main/kotlin/no/nav/su/se/bakover/domain/Sak.kt

```diff
@@ -532,8 +544,70 @@ data class Sak(
             .singleOrNull { it.søknad.id == søknadId }?.right() ?: FantIkkeSøknadsbehandlingForSøknad.left()
     }
 
+    fun opprettNySøknadsbehandling(
+        søknad: Søknad.Journalført.MedOppgave,
+        clock: Clock,
+    ) = if (harÅpenSøknadsbehandling() || hentÅpneRevurderinger().isRight() || hentÅpneReguleringer().isRight()) {
+        HarÅpenBehandling.left()
+    } else
+        NySøknadsbehandling(
+            id = UUID.randomUUID(),
+            opprettet = Tidspunkt.now(clock),
+            sakId = this.id,
+            søknad = søknad,
+            oppgaveId = søknad.oppgaveId,
+            fnr = søknad.søknadInnhold.personopplysninger.fnr,
+            avkorting = this.hentUteståendeAvkortingForSøknadsbehandling().fold({ it }, { it }).kanIkke(),
+            sakstype = søknad.type,
+        ).right()
+
+    fun opprettNyRevurdering(
+        gjeldendeVedtaksdata: GjeldendeVedtaksdata,
+        gjeldendeVedtakPåFraOgMedDato: VedtakSomKanRevurderes,
```

Comment:
Samme som for søknadsbehandling.. veldig rart at vi må sende inn gjeldende vedtaksdata til sak. Litt av poenget her er vel at det er sak som er kilden til disse dataene og følgelig burde eie dette.